### PR TITLE
SWC-6627 - Extend EntityHeaderTable for use in view scope editor

### PR DESCRIPTION
**Pull request by nickgros**
- Allow hiding text field for entering values via prop
 - Allow overriding entity finder config
 - Allow overriding "entity" copy
 - Fix eslint warnings

### CODE REVIEW

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx [hunk 1]
```diff
@@ -41,17 +37,34 @@ import { EntityHeaderTypeCell } from './EntityHeaderTableCellRenderers'
...
 export type EntityHeaderTableProps = {
   references: ReferenceList
   isEditable: boolean
   onUpdate?: (updatedRefs: ReferenceList) => void // when the references are updated, EntityHeaderTable will call this function with the updated list
   removeSelectedRowsButtonText?: string
   onUpdateEntityIDsTextbox?: (value: string) => void // when the entity IDs text box is updated, this is called
+  /* The word used to describe the items in the table. Default 'entity' */
+  objectNameCopy?: string
+  // If true, the text field where IDs are pasted is hidden, and confirming the entity finder will immediately call `onUpdate`
+  hideTextFieldToPasteValue?: boolean
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:66]: Always showing this text field is a little overwhelming for the scope editor

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx [hunk 2]
```diff
@@ -41,17 +37,34 @@ import { EntityHeaderTypeCell } from './EntityHeaderTableCellRenderers'
...
+  entityFinderConfiguration?: EntityFinderModalProps['configuration']
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:67]: In the scope editor, I only want to allow picking containers, so I need to override the finder config

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx [hunk 3]
```diff
@@ -41,17 +37,34 @@ import { EntityHeaderTypeCell } from './EntityHeaderTableCellRenderers'
...
+}
 
 export type EntityHeaderTableProps = {
   references: ReferenceList
   isEditable: boolean
   onUpdate?: (updatedRefs: ReferenceList) => void // when the references are updated, EntityHeaderTable will call this function with the updated list
   removeSelectedRowsButtonText?: string
   onUpdateEntityIDsTextbox?: (value: string) => void // when the entity IDs text box is updated, this is called
+  /* The word used to describe the items in the table. Default 'entity' */
+  objectNameCopy?: string
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:64]: In copy, I want to replace "entity" with "container"

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx [hunk 4]
```diff
@@ -85,9 +101,12 @@ export const EntityHeaderTable = (props: EntityHeaderTableProps) => {
...
   } = useEntityHeaderTableState(references, onUpdateEntityIDsTextbox, onUpdate)
 
-  const setInvalidEntityIDError = useCallback((invalidEntityIDs: string[]) => {
-    setParseErrors([`Invalid Synapse ID(s): ${invalidEntityIDs.join(',')}`])
-  }, [])
+  const setInvalidEntityIDError = useCallback(
+    (invalidEntityIDs: string[]) => {
+      setParseErrors([`Invalid Synapse ID(s): ${invalidEntityIDs.join(',')}`])
+    },
+    [setParseErrors],
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:108]: ESLint complained about missing dependencies--these are all `useCallback` so it is appropriate to include all dependencies

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx [hunk 5]
```diff
@@ -255,21 +282,7 @@ export const EntityHeaderTable = (props: EntityHeaderTableProps) => {
...
-    EntityType.ENTITY_VIEW,
-    EntityType.DOCKER_REPO,
-    EntityType.SUBMISSION_VIEW,
-    EntityType.DATASET,
-    EntityType.DATASET_COLLECTION,
-    EntityType.MATERIALIZED_VIEW,
-    EntityType.VIRTUAL_TABLE,
-  ]
-
+  ;``
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:285]:
```suggestion

```

Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:285]: Took me a second, but I agree with the suggested change of removing this :D

**Review by jay-hodgson — APPROVED** [review]

File: packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx (hunk 5 above)
Comment (nickgros) [reply packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx:285]: I'm surprised ESLint didn't have a problem with this!

**Pull request merged**